docs/service-bundle.md

```diff
@@ -0,0 +1,225 @@
+* JSON: a json document of information that is useful to the bundle. Currently
+  it is expressed in a form that is useful for populating variables in ansible
+  that can be used by roles.
+
+### Actions
+
+Possible actions include
+* provision
+* deprovision
+* bind
+* unbind
+* update
+* test
+
+### JSON
+Example of JSON document passed to __provision__:
+
+```
+{
+    "cluster": "openshift",
+    "_apb_plan_id": "default",
+    "_apb_service_class_id": "c23ec213bb8dea1577230c5ce005b9c2",
+    "_apb_service_instance_id": "54636ad3-0378-49a1-a494-f97d0a0daf8e",
+    "mediawiki_admin_pass": "pass",
+    "mediawiki_admin_user": "admin",
+    "mediawiki_db_schema": "mediawiki",
+    "mediawiki_site_lang": "en",
+    "mediawiki_site_name": "MediaWiki",
+    "namespace": "bazproject"
+}
+```
+
+* cluster: the type of cluster on which the service bundle is being run
+  (currently either ‘openshift’ or ‘kubernetes’).
+* _apb_plan_id: the name of a “plan”, as defined through the OSB API, that is
+  available on the service class.
+* _apb_service_class_id: the “service_id”, as defined through the OSB API, that
+  is being used for the current action.
+* _apb_service_instance_id: the “instance_id”, as defined through the OSB API,
+  that uniquely identifies the service instance being acted upon.
+* namespace: the target namespace in which resources should be created,
+  deleted, or acted upon.
+
+Other keys represent parameter names from the Plan, the values for which have
+the type specified in the Plan.
+
+Example of JSON document passed to __bind__:
+
+```
+{
+    "_apb_plan_id": "dev",
+    "_apb_provision_creds": {
+        "DB_ADMIN_PASSWORD": "yiX2hXH7RU24RTVRXuye",
+        "DB_HOST": "postgresql",
+        "DB_NAME": "foo",
+        "DB_PASSWORD": "letmein",
+        "DB_PORT": "5432",
+        "DB_TYPE": "postgres",
+        "DB_USER": "bar"
+    },
+    "_apb_service_class_id": "1dda1477cace09730bd8ed7a6505607e",
+    "_apb_service_instance_id": "d6ac6b10-fbff-4944-8e6b-3478313e20d1",
+    "cluster": "kubernetes",
+    "namespace": "default"
+}
+```
+
+* _apb_provision_creds: A set of credentials and related data created during a
+  provision action for this service instance. Details are below in the Output
+  section.
+
+### Environment Variables
+
+The following environment variables are set by the broker:
+
+* __POD_NAMESPACE__: the namespace in which the service bundle is being run
+* __POD_NAME__: the name of the pod 
+
+## Output
+
+### stdout / stderr
+
+These are ignored.
+
+### Exit Code
+
+* 0 - success
+* 1 - error
+* 8 - action not implemented/supported
+
+### Encoded Credentials
+
+Binding credentials may be returned during a provision or bind operation by
+creating a named secret (POD_NAME) in the sandbox namespace (POD_NAMESPACE).
+The secret should have a single key called “fields” whose value is a
+base64-encoded JSON document.
+
+Ansible Playbook Bundles use the “asb_encode_binding” module to do this.
```

Comment:
Again, language here is not generic.

Reply:
Here I was intentionally ansible-specific to call out an example of how this is done in practice. If someone wants to see this taking place, they can look for this ansible module being used in an APB.

Reply:
ACK, makes sense. I'd say my comments re: the generic nature of this remain concerns of mine, but we're meeting to tackle that problem specifically, so for the purposes of this PR it's safe to ignore.